pkg/globalnet/controllers/controllers_suite_test.go

```diff
@@ -114,6 +116,10 @@ func newTestDriverBase() *testDriverBase {
 	Expect(submarinerv1.AddToScheme(t.scheme)).To(Succeed())
 	Expect(corev1.AddToScheme(t.scheme)).To(Succeed())
 
+	// TODO: Remove this workaround for https://github.com/kubernetes/client-go/issues/949 once
+	// admiral has been updated
+	t.scheme.AddKnownTypeWithName(schema.GroupVersionKind{Group: "fake-dynamic-client-group", Version: "v1", Kind: "List"}, &unstructured.UnstructuredList{})
+
```

Comment:
I think we should pin to K8s 0.21.11 to avoid having to do this. The rest of the submariner projects use 0.21.11 so I think we should be consistent.

Reply:
For example when I pin to 0.21.11 

```
#14 29.60 # github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops
#14 29.60 vendor/github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops/transact.go:21:20: undefined: wait.PollImmediateUntilWithContext
```

Reply:
> Also something to keep in mind... We don't use `logr` in this repo so that's not a problem anymore... 

But I intend to use `logr`, specifically for zerolog.

> But I do need to ensure that we can import those ovn-k libraries without depending on k8s version.. Or else then every time ovn-k bumps their K8s version we'll have to, creating an unnecessary dependency.

I see that you're using the `master` branch from `github.com/ovn-org/ovn-kubernetes`. Not sure how stable that is but I guess we have to since no prior version has a `go.mod` file.

Reply:
`wait.PollImmediateUntilWithContext ` was added in K8s 0.22 so perhaps try pinning to 0.22.9, which uses `k8s.io/klog/v2` v2.9.0 and `logr` v0.4.0.

Reply:
> I see that you're using the master branch from github.com/ovn-org/ovn-kubernetes. Not sure how stable that is but I guess we have to since no prior version has a go.mod file.

Right they have no upstream release versioning process so it is a bit unstable, but these bits were already pretty unstable, and regardless I don't think what we have here is necessarily the best long-term solution since any way we cut it there will always be a tightly coupled dependency with the CNI version

Reply:
Ideally we would bump to K8s 0.23 across the board. At one point **lighthouse** was using 0.23 due to `coredns` bump but we downgraded due to **submariner-operator** issues.  Obviously **submariner** is building fine with 0.23 with this PR.  I'm not sure what's holding us back. @skitt? Is it the older operator framework in **submariner-operator**?

Reply:
I tried pinning to K8s 0.22.9 et al but hit a showstopper:

```
github.com/submariner-io/submariner/pkg/networkplugin-syncer/handlers/ovn imports
        github.com/ovn-org/libovsdb/client imports
        github.com/go-logr/stdr imports
        github.com/go-logr/logr/funcr: no required module provides package github.com/go-logr/logr/funcr; to add it:
        go get github.com/go-logr/logr/funcr
~/go/src/github.com/submariner-io/submariner>go get github.com/go-logr/logr/funcr
go: module github.com/go-logr/logr@upgrade found (v1.2.3, replaced by github.com/go-logr/logr@v0.4.0), but does not contain package github.com/go-logr/logr/funcr
```

So `github.com/go-logr/logr/funcr` was added in `logr` v1.0.0 which also introduces the backwards incompatible `Logger` change from interface to struct. So in order to use `libovsdb`, we need K8s 0.23 et al. We'll see what happens when the **submariner** version is bumped in **submariner-operator** 😄 Of course bumping everything to K8s 0.23 would help.

Reply:
Yeah I experienced the exact same thing :(  

And sounds good, I'll let this sit for now since CI is green and continue to monitor the best way forward